src/deploy/index.ts

```diff
@@ -69,7 +69,17 @@ export const deploy = async function (
 
   if (targetNames.includes("hosting")) {
     const config = options.config.get("hosting");
-    if (Array.isArray(config) ? config.some((it) => it.source) : config.source) {
+    if (
+      Array.isArray(config)
+        ? config.some(
+            (it) =>
+              it.source &&
+              (!options.only.includes("hosting:") ||
+                new RegExp(`\\bhosting:${it.target}\\b`).exec(options.only)),
```

Comment:
I understand this is lame, and reflects my lack of understanding of all the config flags/etc and how best to parse this. But the point is to make this logic, "Enforce webframeworks if either all sites need to be deployed, or if the site being deployed itself has `source`").

Reply:
@fivecar I just pushed [this commit](https://github.com/firebase/firebase-tools/pull/8169/commits/00cc1fc2183bb62f31cb16928e438a4f6cb2d503) to improve readability and to handle the case when using `firebase deploy` which was erroring out.
All the functionality should remain the same.

Reply:
Awesome - it looks so much better, @chalosalvador . Thanks for your help and support!